### PR TITLE
Ch5 issue124

### LINKED ISSUE #124: What are some of the ways to do EDA for CV tasks?
**Section Outline**
- Checks on images generally
- EDA Tasks
- Bounding Box Formats
- Bounding Box Random Sample Checks
- Masks Sample Checks

 **Definition of Done**
- [x] Completed `5-data-mgmt-exp-proc/eda-cv-tasks.md`
- [x] Verified against [self-review checklist](https://github.com/aisingapore/ai-practitioner-handbook/blob/main/REVIEWING.md)
- [ ] Added section to [table of contents](https://github.com/aisingapore/ai-practitioner-handbook/blob/main/_toc.yml)
- [ ] Page builds and displays correctly on local machine

### PR DESCRIPTION
**Pull request by tappyness1**
EDA for CV Tasks

### CODE REVIEW

**Comment by ngkh2018** [comment]
Under 'Conclusion', it mentions that 'Engineer should explore further methods for EDA'. Are the topic listing under 'References' the methods an engineer can refer to for exploration? If not, it would be good to suggest a list of methods and the corresponding URL links.

Another trivial comment is to change the 2 following sub headers under 'Image EDA', so as to be consistent in naming the sub headers
- 'Brightness' to 'Brightness of Image'
- 'Contrast' to 'Contrast of Image'

**Review by najibninaba — APPROVED** [review]
No further comments from me.

File: book/5-data-mgmt-exp-proc/eda-cv-tasks.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+## Image EDA  
+
+### Basic Image Analysis  
+The basic image analysis considers the broad characteristics of the images that have been obtained.
+
+The guiding questions are as such:   
+1. What is the number of images provided? Is this the number of images promised to be delivered for experiments?
+1. What is the format of images eg JPG/PNG/JPEG? If there are multiple formats given, what are the counts for each image format? 
+1. What are the image sizes/resolution (eg 1080x960)? If there are multiple dimensions, what are the counts for each dimension?
+1. How similar are the images to each other? You can do an EDA to check on this by performing image hashing to discover groups which are similar to each other.
```
Comment (kevinosl) [inline book/5-data-mgmt-exp-proc/eda-cv-tasks.md]: Suggest adding this to point 4:
`This also helps to uncover whether some of the images might have been curated from a continuous video stream of a scene that is changing very slowly - in such a case, these images may be near-duplicates.`

File: book/5-data-mgmt-exp-proc/eda-cv-tasks.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+## Annotations Random Sample Checks
+
+### Bounding Box and Mask Random Sample Checks
+
+Bounding Boxes Random Sample Check refers to performing a check by sampling images and analysing their annotations. These try to account to account for human error that very often occurs. 
+
+1. Are the bboxes/masks drawn tight enough? 
+1. Were the bboxes/masks drawn accurately?
+1. Are the objects being occluded? In other words, were parts of the object blocked by another object?
```
Comment (kevinosl) [inline book/5-data-mgmt-exp-proc/eda-cv-tasks.md]: How about adding this pointer to point 3?
`In general, if the object of interest is being occluded, its annotation should still be drawn such that it includes the occluded portion.`

File: book/5-data-mgmt-exp-proc/eda-cv-tasks.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+### Basic Annotations Analysis
+
+The basic annotations analysis attempts to find characteristics of the annotation such as number (and proportion) of classes as well as the values of the bounding boxes/ masks. 
+
+This is useful because it helps determine what type of metrics should be used, how the data (images) should be split. For metrics, the usual case for this is error analysis of model performance on bounding box sizes. For example, if the bboxes/masks are mostly large objects among all the images, assuming images are sufficiently clear, it is clear that the model will do well. However, if the annotations are small, then the model will inevitably struggle. Hence, the AI Engineer will pay attention to certain metrics more, such as mAP with Area=Large.
+
+The data split refers to how one might partition the data for training/validating/testing the model. Generally, all sets will need equal proportions of the bbox size (small/medium/large) and class distribution. In the ideal scenario, these proportions are easy to split at random, but it is extremely rare. Thus, by performing EDA on the annotations, the Engineer will have a better idea on how the data split can be done. 
+
```
Comment (kevinosl) [inline book/5-data-mgmt-exp-proc/eda-cv-tasks.md:72]: Suggest adding this benefit as well?
`Knowing the height and width distribution of the bboxes also helps to inform the range of anchor box sizes to be considered during model training (for models that require it).`

File: book/5-data-mgmt-exp-proc/eda-cv-tasks.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+With these considerations in mind, these are some guiding questions when performing EDA on annotations: 
+1. How many instances of each classes are there?
+1. Bounding Box - what is width and height distribution overall and per class?
+1. Bounding Box - what is the overall size (W*H) overall and per class?
+1. Bounding Box - What are the top few Width/Height combination? This can be found using clustering methods.
+1. Segmentation - what is distribution of the mask pixel counts overall and per class?
+
```
Comment (kevinosl) [inline book/5-data-mgmt-exp-proc/eda-cv-tasks.md:79]: Not sure if space permits - if so, may be nice to put in an example of the usual height vs width scatterplot, which is a very useful visual aid. We can use one of the examples from our projects, without mentioning the project.

File: book/5-data-mgmt-exp-proc/eda-cv-tasks.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+### Basic Annotations Analysis
+
+The basic annotations analysis attempts to find characteristics of the annotation such as number (and proportion) of classes as well as the values of the bounding boxes/ masks. 
+
+This is useful because it helps determine what type of metrics should be used, how the data (images) should be split. For metrics, the usual case for this is error analysis of model performance on bounding box sizes. For example, if the bboxes/masks are mostly large objects among all the images, assuming images are sufficiently clear, it is clear that the model will do well. However, if the annotations are small, then the model will inevitably struggle. Hence, the AI Engineer will pay attention to certain metrics more, such as mAP with Area=Large.
+
+The data split refers to how one might partition the data for training/validating/testing the model. Generally, all sets will need equal proportions of the bbox size (small/medium/large) and class distribution. In the ideal scenario, these proportions are easy to split at random, but it is extremely rare. Thus, by performing EDA on the annotations, the Engineer will have a better idea on how the data split can be done.
```
Comment (kevinosl) [inline book/5-data-mgmt-exp-proc/eda-cv-tasks.md]: Do you mean to say all sets will need similar proportions of the bbox size (as opposed to equal)?

**Review by kevinosl — COMMENTED** [review]
Great article Calvin. Its practical and I can see it being very useful to a new engineer. I've left some specific comments inline.

File: book/5-data-mgmt-exp-proc/eda-cv-tasks.md (hunk 3 above)
Comment (tappyness1) [reply book/5-data-mgmt-exp-proc/eda-cv-tasks.md:72]: Added a paragraph to talk about how the distribution can inform the type of parameters to be tuned, with anchor box sizes as an example.

**Pull request merged**